nomad/data_source_scheduler_config.go

```diff
@@ -0,0 +1,54 @@
+package nomad
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceSchedulerConfig() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceSchedulerConfigRead,
+
+		Schema: map[string]*schema.Schema{
+			"scheduler_algorithm": {
+				Description: "Specifies whether scheduler binpacks or spreads allocations on available nodes.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"preemption_config": {
+				Description: "Options to enable preemption for various schedulers.",
+				Computed:    true,
+				Type:        schema.TypeMap,
+				Elem:        &schema.Schema{Type: schema.TypeBool},
+			},
+		},
+	}
+}
+
+func dataSourceSchedulerConfigRead(d *schema.ResourceData, meta interface{}) error {
+
+	client := meta.(ProviderConfig).client
+
+	schedCfg, _, err := client.Operator().SchedulerGetConfiguration(nil)
+	if err != nil {
+		return fmt.Errorf("failed to query scheduler config: %v", err)
+	}
```

Comment:
@lgfa29 added a StateWriter utility on Friday that helps with these error checks: 
https://github.com/hashicorp/terraform-provider-nomad/pull/162/files#diff-3bd71acc5dfea73f22ad03c6d60d4f85ab09c8a1d818a1c4c148cfac12c5eab0R71-R86